tests/containers/kubectl.pm

```diff
@@ -0,0 +1,139 @@
+    assert_script_run('mkdir -p ~/.kube');
+    my $provider = get_var("KUBECTL_CLUSTER", "k3s");
+    if ($provider eq "k3s") {
+        install_k3s();
+
+        # Ensure k3s has not installed its own kubectl
+        assert_script_run("! stat /usr/local/bin/kubectl");
+        validate_script_output("whereis kubectl", sub { $_ !~ m/\/usr\/local\/bin/ });
+    } else {
+        die "Invalid or unsupported KUBECTL_CLUSTER";
+    }
+
+    # Function to check if the output either has no resources or has at least two lines
+    # Two lines because one line is the header and the following lines are the data rows
+    validate_script_output("kubectl cluster-info", qr/Kubernetes control plane/);
+    record_info('Test get', 'State tests (get commands)');
+    # Check for default namespaces to be present
+    validate_script_output("kubectl get namespaces", qr/default/);
+    validate_script_output("kubectl get namespaces", qr/system/);
+    # Check service output
+    validate_script_output("kubectl get service --all-namespaces", sub { $_ =~ m/No resources found/ || split(/\n/, $_) > 1; });
+    # At least the coredns pod must be present
+    validate_script_output("kubectl get pods --all-namespaces", sub { $_ =~ m/coredns/ });
+    # There must be at least one endpoint present
+    validate_script_output("kubectl get endpoints", sub { split(/\n/, $_) > 1; });
+    # There must be at least the coredns deployment present
+    validate_script_output("kubectl get deployments --all-namespaces", sub { $_ =~ m/coredns/ });
+    # Check if more than 1 replicasets are present
+    validate_script_output("kubectl get replicasets --all-namespaces --no-headers", sub { split(/\n/, $_) > 1; });
+    # Check ingresses, those are empty by default
+    validate_script_output("kubectl get ingresses --all-namespaces --no-headers", sub { $_ =~ m/No resources found/ });
+
+    ## Test the configuration
+    record_info('Test config', 'Configugration tests');
+    assert_script_run("kubectl config view");
+    validate_script_output("kubectl config view -o jsonpath={.clusters[].name}", qr/default/);
+    validate_script_output("kubectl config view -o jsonpath={.contexts[].name}", qr/default/);
+
+    ## Test the context configuration
+    assert_script_run("kubectl config get-contexts");
+    validate_script_output("kubectl config get-contexts", sub { $_ =~ m/default/ });
+    validate_script_output("kubectl config current-context", sub { $_ =~ m/default/ });
+    assert_script_run("kubectl config use-context default");
+    assert_script_run("kubectl config set-context --current --namespace=default");
+    assert_script_run("kubectl config set-context localhost --user=root --namespace=default");
+    assert_script_run("kubectl config use-context localhost");
+    assert_script_run("kubectl config use-context default");
+    assert_script_run("kubectl config unset contexts.localhost");
+
+    ## Test jobs
+    record_info('Test 3', 'job and pod test runs');
```

Comment:
You name this "test 3" but the other tests don't have a number. Either use numbers everywhere or just use here "Test jobs" to be more consistent.